ManagementSystem/PhoneBook.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const PersonalContact = require('./PersonalContact');
+const BusinessContact = require('./BusinessContact');
+const { ContactNotFoundError, FileSystemError } = require('./Errors');
+
+const fileName = 'contacts.txt';
+
+class PhoneBook {
+    constructor() {
+        this.contacts = [];
+        this.loadContactsFromFile();
+    }
+
+    addContact(contact) {
+        this.contacts.push(contact);
+        this.saveContactsToFile();
+    }
+
+    updateContact(name, newContact) {
+        const index = this.contacts.findIndex(contact => contact.name === name);
+        if (index !== -1) {
+            this.contacts[index] = newContact;
+            this.saveContactsToFile();
+        } else {
+            throw new ContactNotFoundError(`Contact name "${name}" not found.`);
+        }
+    }
+
+    deleteContact(name) {
+        const index = this.contacts.findIndex(contact => contact.name === name);
+        if (index !== -1) {
+            this.contacts.splice(index, 1);
+            this.saveContactsToFile();
+        } else {
+            throw new ContactNotFoundError(`Contact name "${name}" not found.`);
+        }
+    }
+
+    displayContacts() {
+        if (this.contacts.length === 0) {
+            console.log("No contacts are available.");
+        } else {
+            this.contacts.forEach(contact => contact.display());
```

Comment:
How are you able to display contact from the file here?

Reply:
Display method is used in the app.js file when the user chooses to display contacts.

Reply:
How are you reading data from file here?